File: app/src/main/java/oleg/osipenko/mai/data/repository/specification/ListContentSpecification.java
package oleg.osipenko.mai.data.repository.specification;

/**
 * Created by olegosipenko on 07.09.15.
 */
public class ListContentSpecification implements RequestSpecification<String> {

    private String menuItem;

    public ListContentSpecification(String menuItem) {
        this.menuItem = menuItem;
    }

    @Override
    public boolean specified(String string) {
        return string.equals(menuItem);
    }

    public String getParameter() {
        return menuItem;
    }
}

File: app/src/main/java/oleg/osipenko/mai/data/repository/specification/ListContentSpecification.kt
package oleg.osipenko.mai.data.repository.specification

import oleg.osipenko.mai.data.repository.RequestSpecification

/**
 * Created by olegosipenko on 05.09.15.
 */
public class ListContentSpecification : RequestSpecification<String> {

    override fun specified(s: String): Boolean {
        return false
    }
}
